src/commands/cmd_hll.cc

```diff
@@ -75,7 +75,29 @@ class CommandPfCount final : public Commander {
   }
 };
 
+/// PFMERGE destkey [sourcekey [sourcekey ...]]
+///
+/// complexity: O(N) to merge N HyperLogLogs, but with high constant times.
+class CommandPfMerge final : public Commander {
+  Status Execute(Server *srv, Connection *conn, std::string *output) override {
+    redis::HyperLogLog hll(srv->storage, conn->GetNamespace());
+    std::vector<std::string> keys(args_.begin() + 1, args_.end());
+    std::vector<Slice> dest_keys;
+    dest_keys.reserve(args_.size() - 1);
+    for (size_t i = 1; i < args_.size(); i++) {
+      dest_keys.emplace_back(args_[i]);
+    }
+    auto s = hll.Merge(args_[0], dest_keys);
+    if (!s.ok() && !s.IsNotFound()) {
+      return {Status::RedisExecErr, s.ToString()};
+    }
+    *output = redis::SimpleString("OK");
+    return Status::OK();
+  }
+};
+
 REDIS_REGISTER_COMMANDS(MakeCmdAttr<CommandPfAdd>("pfadd", -2, "write", 1, 1, 1),
-                        MakeCmdAttr<CommandPfCount>("pfcount", 2, "read-only", 1, 1, 1), );
+                        MakeCmdAttr<CommandPfCount>("pfcount", 2, "read-only", 1, 1, 1),
+                        MakeCmdAttr<CommandPfMerge>("pfmerge", -2, "write", 1, 1, 1), );
```

Comment:
We cannot use the static key range for PFMERGE/PFADD, because from 1..N are keys for those two commands.

Reply:
I think it can be `1, -1, 1`?

Reply:
oh, good call. Should can use `1, 1, 0` and it will iterate to the end if the last key index is <= 0:

```C++
  template <typename F>
  void ForEachKey(F &&f, const std::vector<std::string> &args) const {
    for (size_t i = first_key; last_key > 0 ? i <= size_t(last_key) : i <= args.size() + last_key; i += key_step) {
      std::forward<F>(f)(args[i]);
    }
  }
```

Reply:
Why `1, 1, 0`?

IMHO first key = 1, last key = -1, key step = 1 here.